controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -757,6 +782,45 @@ func saveHostProvisioningSettings(host *metal3v1alpha1.BareMetalHost) (dirty boo
 		dirty = true
 	}
 
+	// Copy RAID settings
+	if host.Spec.RAID != host.Status.Provisioning.RAID {
+		// If RAID settings is nil, remove saved settings,
+		// else check hardware RAID and software RAID.
+		if host.Spec.RAID == nil {
+			host.Status.Provisioning.RAID = nil
+			dirty = true
+		} else {
+			if host.Status.Provisioning.RAID == nil {
+				host.Status.Provisioning.RAID = &metal3v1alpha1.RAIDConfig{}
+				dirty = true
+			}
+			// If HardwareRAIDVolumes isn't nil, we will ignore SoftwareRAIDVolumes.
```

Comment:
Can we issue an error when both are set?

Reply:
Errors are not really a thing(!). The place to do that would be in a validating webhook (which runs before the bad spec is stored - here it is after, aka too late), but we don't have one yet.

Reply:
SIGH. Can we at least log an ugly warning?

Reply:
If you want to output the log here, it seems that we need to add a new parameter.

Reply:
Let's research this option in a follow-up.

Reply:
Can I add the warning to func [setTargetRAIDCfg](https://github.com/metal3-io/baremetal-operator/pull/292/files#diff-d3542d6731fdd4d0ef86ed2bc8f7bbad22a4a0af2b8c28d0107835a32f7afce1R15)? Because when the state returns from ready to preparing, this warning seems to be output twice in a row.